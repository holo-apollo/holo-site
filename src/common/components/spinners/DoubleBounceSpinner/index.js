import React from 'react';
import styled, { css, keyframes } from 'styled-components';

import palette from 'common/palette';

const skBounce = keyframes`
  0%, 100% { transform: scale(0.0); }
  50% { transform: scale(1.0); }
`;

const getSpinnerStyles = ({ size }) => css`
  width: ${size}px;
  height: ${size}px;
`;

const Spinner = styled.div`
  position: relative;

  ${getSpinnerStyles};
`;

const getDoubleBounceStyles = ({ color, index }) => css`
  background-color: ${color};
  ${index === 1 && 'animation-delay: -1.0s;'}
`;

const DoubleBounce = styled.div`
  width: 100%;
  height: 100%;
  border-radius: 50%;
  opacity: 0.6;
  position: absolute;
  top: 0;
  left: 0;
  animation: ${skBounce} 2s infinite ease-in-out;

  ${getDoubleBounceStyles};
`;

const DoubleBounceSpinner = ({ color, size }) => (
  <Spinner size={size}>
    <DoubleBounce color={color} index={0} />
    <DoubleBounce color={color} index={1} />
  </Spinner>
);

DoubleBounceSpinner.defaultProps = {
  color: palette.blue,
  size: 40,
};

export default DoubleBounceSpinner;
