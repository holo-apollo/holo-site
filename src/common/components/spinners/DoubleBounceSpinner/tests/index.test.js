import React from 'react';
import renderer from 'react-test-renderer';

import DoubleBounceSpinner from '../index';

describe('DoubleBounceSpinner', function() {
  it('should match snapshot', function() {
    const tree = renderer.create(<DoubleBounceSpinner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
