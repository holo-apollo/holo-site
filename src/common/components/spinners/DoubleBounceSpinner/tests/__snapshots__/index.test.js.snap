// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoubleBounceSpinner should match snapshot 1`] = `
<div
  className="DoubleBounceSpinner__Spinner-kwvmmk-0 gMRmkM"
  size={40}
>
  <div
    className="DoubleBounceSpinner__DoubleBounce-kwvmmk-1 dIQfd"
    color="#4caad0"
  />
  <div
    className="DoubleBounceSpinner__DoubleBounce-kwvmmk-1 fxFNSh"
    color="#4caad0"
  />
</div>
`;
